Yoast/wordpress-seo
# Upgrade Jest to latest version

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Need the latest version of Jest to have more specific matcher assertions available for testing.

## Test instructions

This PR can be tested by following these steps:

* `Yarn install`, then `yarn test`.
* Verify that the tests still work.

## Quality assurance

~* [ ] I have added unittests to verify the code works as intended~
